internal/registry/registry.go

```diff
@@ -40,24 +43,24 @@ type RepoPullConfig struct {
 //go:generate mockgen -source=registry.go -package=registry -destination=mock_registry_api.go
 
 type Registry interface {
-	ImageExists(ctx context.Context, image string, po *kmmv1beta1.PullOptions, registryAuthGetter auth.RegistryAuthGetter) (bool, error)
+	ImageExists(ctx context.Context, image string, po *kmmv1beta1.PullOptions, secret *corev1.LocalObjectReference, namespace string) (bool, error)
 	VerifyModuleExists(layer v1.Layer, pathPrefix, kernelVersion, moduleFileName string) bool
-	GetLayersDigests(ctx context.Context, image string, registryAuthGetter auth.RegistryAuthGetter) ([]string, *RepoPullConfig, error)
+	GetLayersDigests(ctx context.Context, image string, secret *corev1.LocalObjectReference, namespace string) ([]string, *RepoPullConfig, error)
 	GetLayerByDigest(digest string, pullConfig *RepoPullConfig) (v1.Layer, error)
 }
 
-type registry struct{}
-
-func NewRegistry() Registry {
-	return &registry{}
+type registry struct {
```

Comment:
I know we agreed on that but I know see the implications of it - don't you think we should detach the `registry` interface from the k8s cluster itself. 

I am not a fan of making the registry API tied to the k8s client.

WDYT?